client/components/room/Video.jsx

```diff
@@ -27,192 +33,285 @@ class Video extends React.Component {
     if (event.candidate) {
-      this.getOtherPc(pc).addIceCandidate(
-        new RTCIceCandidate(event.candidate)
-      ).then(
-        () => {
-          console.log(`${this.getName(pc)} addIceCandidate success!`);
-        },
-        (error) => {
-          console.error(`${this.getName(pc)} failed to add ICE Candidate: ${error.toString()}`);
-        }
-      );
-      console.log(`${this.getName(pc)} ICE candidate: ${event.candidate.candidate}`);
+      this.sendMessage({
+        type: 'candidate',
+        label: event.candidate.sdpMLineIndex,
+        id: event.candidate.sdpMid,
+        candidate: event.candidate.candidate
+      });
+    } else {
+      console.log('End of candidates.');
     }
   }
 
-  gotRemoteStream(event) {
+  handleRemoteStreamAdded(event) {
     window.remoteStream = remoteVideo.srcObject;
     remoteVideo.srcObject = event.stream;
-    console.log('The pc2 received remote stream');
+    console.log('Remote stream added');
   }
 
-  getName(pc) {
-    return (pc === pc1) ? pc1 : pc2;
+  handleRemoteStreamRemoved(event) {
+    console.log(`Remote stream removed. Event: ${event}`);
   }
 
-  getOtherPc(pc) {
-    return (pc === pc1) ? pc2 : pc1;
+  /////////////////////////////////
+
+  hangup() {
+    console.log('Hanging up.');
+    this.stop();
+    sendMessage('bye');
   }
 
-  handleVideoScreen() {
+  handleRemoteHangup() {
+    console.log('Session terminated.');
+    this.stop();
     this.setState({
-      constraints: {
-        video: !this.state.constraints.video,
-        audio: this.state.constraints.audio
-      },
-      video: !this.state.constraints.video ? 'Video Off' : 'Video On'
-    }, () => (this.start()));
+      isInitiator: false
+    });
   }
 
-  handleAudio() {
+  stop() {
     this.setState({
-      constraints: {
-        video: this.state.constraints.video,
-        audio: !this.state.constraints.audio
-      },
-      mute: !this.state.constraints.audio ? 'Mute' : 'Unmute'
-    }, () => (this.start()));
-    navigator.mediaDevices.getUserMedia(this.state.constraints)
-      .then(this.successCallback)
-      .catch(this.errorCallback);
+      isStarted: false
+    });
+    pc.close();
+    pc = null;
   }
 
+  // handleVideoScreen() {
+  //   this.setState({
+  //     constraints: {
+  //       video: !this.state.constraints.video,
+  //       audio: this.state.constraints.audio
+  //     },
+  //     video: !this.state.constraints.video ? 'Video Off' : 'Video On'
+  //   }, () => (this.start()));
+  // }
+
+  // handleAudio() {
+  //   this.setState({
+  //     constraints: {
+  //       video: this.state.constraints.video,
+  //       audio: !this.state.constraints.audio
+  //     },
+  //     mute: !this.state.constraints.audio ? 'Mute' : 'Unmute'
+  //   }, () => (this.start()));
+  //   navigator.mediaDevices.getUserMedia(this.state.constraints)
+  //     .then(this.successCallback)
+  //     .catch(this.errorCallback);
+  // }
+
```

Comment:
Same thing as with the function bindings. Wondering if we can remove this commented block before merging.